app/clients/erddap.js

```diff
@@ -36,15 +36,24 @@ const getSingleBuoyGeoJsonData = ({
   const startDate = start ?? "2003-01-01T12:00:00Z";
   const endDate = end ?? "2012-12-31T12:00:00Z";
   idString = idString ?? `"${id}"`;
+  const queryString = `/${datasetId}.geoJson?${variables.join(
+    ","
+  )},${baseVariables.join(
+    ","
+  )}&station_name=${idString}&time>=${startDate}&time<=${endDate}`;
```

Comment:
Why build this by hand when you can use `URLSearchParams`!?

https://nodejs.org/dist/latest-v14.x/docs/api/url.html#url_class_urlsearchparams
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

```js
const params = new URLSearchParams()
params.append('newKey', 'value')
console.log(params.toString())
```

Reply:
oooh - that does seem nicer

Reply:
So it turns out ERDDAP doesn't use standard search params and uses things like `<=` and `>=` and multiple keys, so I'm stuck with string interpolation 😞